README.md

```diff
@@ -61,6 +61,51 @@ You can also always find the full command-line options that are available with
 $ cargo fuzz --help
 ```
 
+## Generating code coverage information
+
+Use the `--coverage` option to generate precise
+[source-based code coverage](https://blog.rust-lang.org/inside-rust/2020/11/12/source-based-code-coverage.html)
+information:
+```
+$ cargo fuzz run --coverage <coverage data output file> <target>
+```
+This compiles your project using the `-Zinstrument-coverage` Rust compiler flag and generates coverage data in the
+specified file. This file can be used to generate coverage reports and visualize code-coverage information.
+If you run the fuzzer multiple times, you can specify different coverage-output file names and subsequently merge them
+into one data file.
+Read more in the [Unstable book](https://doc.rust-lang.org/beta/unstable-book/compiler-flags/source-based-code-coverage.html#installing-llvm-coverage-tools).
+
+### Example
+
+Suppose we have a `compiler` fuzz target for which we want to visualize code coverage.
+
+1. Run the fuzzer on the `compiler` target for 60 seconds:
+   
+   `$ cargo fuzz build --coverage "run1" compiler` -- -max_total_time=60
+   
+   This will generate a file named `run1.profraw` in the same directory.
+
+2. You can run the fuzzer again, generating more profiler data in another output file:
+
+   `$ cargo fuzz run --coverage "run2" compiler` -- -max_total_time=60
+
+3. Merge the coverage data files and index them with the instrumented compiler code: 
+  
+   `$ llvm-profdata merge -sparse run1.profraw run2.profraw -o runs.profdata`
+
+4. Visualize the coverage data in HTML:
+
+   `$ llvm-cov show target/.../compiler --format=html -instr-profile=runs.profdata > index.html`
+   
+   There are many visualization and coverage-report options available (see `llvm-cov show --help`).
+
+Note:
+- We recommend using LLVM 11 and a recent nightly version of the Rust toolchain. 
+  This code was tested with `1.51.0-nightly (2021-02-10)`.
+- Coverage information will be written to the `.profraw` file after `cargo fuzz` successfully finishes running.
+  One way to ensure this is to use the `--max_total_time=<seconds>` or `--runs=<number>` libfuzzer option,
```

Comment:
Is "this" in "ensure this" the successful termination? Would fuzzer finding a crash count as a successful termination?